standalone-metastore/metastore-server/src/test/java/org/apache/hadoop/hive/metastore/TestPartitionManagement.java

```diff
@@ -91,13 +91,19 @@ public void tearDown() throws Exception {
           // First drop any databases in catalog
           List<String> databases = client.getAllDatabases(catName);
           for (String db : databases) {
+            for (String table : client.getAllTables(catName, db)) {
+              client.dropTable(catName, db, table, true, true);
+            }
             client.dropDatabase(catName, db, true, false, true);
```

Comment:
Is it possible that the `cascade` option is not implemented as expected?
https://github.com/apache/hive/blob/rel/release-4.0.1/standalone-metastore/metastore-common/src/main/java/org/apache/hadoop/hive/metastore/IMetaStoreClient.java#L1844-L1845

Reply:
Extremely sorry for the late reply, thanks for your patience. I checked and figured out that there is no issue in at least the `java` code related to `cascade`. As you pointed out, it is highly possible that the non-determinism can be because of the `C` scripts related to `cascade`. I am still checking on that.